internal/utils/flags/flags.go

```diff
@@ -231,7 +241,9 @@ func (f StringSliceFlag) Register(fs *pflag.FlagSet) {
 		fs.StringSliceVarP(f.Value, f.Name, f.Shorthand, f.DefaultValue, f.Usage.String())
 	}
 
-	if f.Hidden {
+	if f.Deprecate != "" {
```

Comment:
I should have thought of this earlier, but now with this becoming slightly more complex (an if + else if), we could move this stuff to a little helper in this file

```go
// in each Register impl, we just call the helper now...
func (f BoolFlag) Register(fs *pflag.FlagSet) {
  // register the flag var

  registerFlag(fs, f.Meta)
}

func registerFlag(fs *pflag.FlagSet, f Meta) {
	if f.Deprecate != "" {
		MarkDeprecated(fs, f.Name, f.Deprecate)
	} else if f.Hidden {
		MarkHidden(fs, f.Name)
	}
}
```